hpx/state.hpp

```diff
@@ -24,11 +24,12 @@ namespace hpx
         state_pre_main = 3,
         state_starting = 4,
         state_running = 5,
-        state_pre_shutdown = 6,
-        state_shutdown = 7,
-        state_stopping    = 8,
-        state_terminating = 9,
-        state_stopped = 10,
+        state_suspended = 6,
```

Comment:
We merged the states for the runtime and the thread manager. For the scheduler to be suspended makes sense for the thread pool executors only. Those 'suspend' by giving back control to the base-scheduler on a regular basis (see idle-callback, [here](https://github.com/STEllAR-GROUP/hpx/blob/master/hpx/runtime/threads/detail/scheduling_loop.hpp#L421), and [here](https://github.com/STEllAR-GROUP/hpx/blob/master/src/runtime/threads/executors/thread_pool_executors.cpp#L277))